app/models/pipeline_run_stage.rb

```diff
@@ -123,26 +132,18 @@ def due_for_aegea_check?
   end
 
   def update_job_status
-    if !id || !started? || succeeded? || failed?
-      # This is called only from PipelineRun.update_status and only when the guard above is false.
+    if !id || !started?
       Airbrake.notify("Invalid precondition for PipelineRunStage.update_job_status #{id} #{job_id} #{job_status}.")
       return
     end
-    if checked?
-      # This can happen due to overly-eager frequent calls from pipeline_monitor through PipelineRun.update_job_status.
-      Rails.logger.info "Job #{id} #{job_id} checked and waiting to load results."
-      return
-    end
-    if output_ready?
-      self.job_status = STATUS_CHECKED
-      # save before enqueue, to prevent minor race
-      save
-      Resque.enqueue(LoadResultForRunStage, id)
+    check_status_file_and_update(JOB_SUCCEEDED_FILE_SUFFIX, STATUS_SUCCEEDED)
+    check_status_file_and_update(JOB_FAILED_FILE_SUFFIX, STATUS_FAILED)
+    if failed? || succeeded?
+      terminate_job
       return
     end
     # The job appears to be in progress.  Check to make sure it hasn't been killed in AWS.   But not too frequently.
     return unless due_for_aegea_check?
-    # db_load_assembly(false) # update the list of taxids that have already been assembled by polling S3
     stdout, stderr, status = Open3.capture3("aegea", "batch", "describe", job_id.to_s)
     unless status.exitstatus.zero?
       Airbrake.notify("Error for update job status for pipeline run #{id} with error #{stderr}")
```

Comment:
In this case you need to update the status as though the jobid.failed file had been created.

Reply:
that's what `self.job_status = STATUS_FAILED if stderr =~ /IndexError/ # job no longer exists` does?